gems/aws-sdk-core/lib/aws-sdk-core/signers/sigv4.rb

```diff
@@ -4,19 +4,34 @@ module AWS::SDK::Core
   module Signers
     # A signer that signs requests using the SigV4 Auth scheme.
     class SigV4 < Hearth::Signers::Base
-      def sign(request:, identity:, properties:)
-        signer = AWS::SigV4::Signer.new
+      # @param signer [AWS::SigV4::Signer] An initialized signer, allowing
+      #   override of default signing parameters. To override default signing
+      #   behavior, configure an auth_scheme on the client:
+      #
+      #     client = Aws::S3::Client.new(
+      #       auth_schemes: [
+      #         AWS::SDK::Core::AuthSchemes::SigV4.new(
+      #           signer: AWS::SDK::Core::Signers::SigV4.new(
+      #            signer: AWS::SigV4::Signer.new(**my_signing_properties)))
+      #       ])
+      #
+      #    Note: If you need to override resolved signing properties, you must
+      #    wrap the auth_resolver and modify the returned properties
+      #    rather than initializing a signer with those properties - providing
+      #    an initialized signer here is only for overriding signing defaults.
```

Comment:
I've updated.

We should have some SRA related documentation on how to do this somewhere, but this probably isn't the place for an example of how to do that.  I mostly wanted to add the docs here on how to override the signer and use it since its for this class.